plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -955,8 +962,9 @@ private void finishOptimize(ConnectorSession session, IcebergTableExecuteHandle
                             .map(dataFile -> dataFile.path().toString())
                             .collect(toImmutableSet()));
         }
+
         RewriteFiles rewriteFiles = transaction.newRewrite();
-        rewriteFiles.rewriteFiles(scannedFiles, newFiles);
+        rewriteFiles.rewriteFiles(scannedDataFiles, scannedDeleteFiles, newFiles, ImmutableSet.of());
```

Comment:
So, Equality Deletes are an edge case we're going to have to deal with here. They're not very common, can't be written by Trino or Spark, but they are special in that they can be global, applying to all partitions.

If we have any of those, they are still safe to delete if the `optimize` did not have any filters applied. If there were filters applied, we should leave the global Equality Deletes alone, since they might apply to other partitions we didn't optimize.

There's an example of how to write one of those here: https://github.com/trinodb/trino/blob/master/plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergV2.java#L355